### PR TITLE
Let secret op create pkcs12 stores

### PR DESCRIPTION
**Pull request by maltesander**
# Description

- removed copying of keystores, now accessed directly vom secret-op moints (nothing is added)
- do not import system trust store anymore
- removed prepare container

fixes https://github.com/stackabletech/kafka-operator/issues/618

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
# Author
- [x] Changes are OpenShift compatible
- [x] CRD changes approved
- [x] Helm chart can be installed and deployed operator works
- [ ] Integration tests passed (for non trivial changes)
- [x] Changes need to be "offline" compatible
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
```

### CODE REVIEW

**Comment by sbernauer** [comment]
I would have excepted much more simplification similar to https://github.com/stackabletech/zookeeper-operator/pull/695,
as in the case of Kafka I *think* we don't need to add additional certs to trust and keystore, so we should be able to straight up mount and use trust and keystores.

**Comment by maltesander** [comment]
test: https://ci.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/kafka-operator-it-custom/48/

File: rust/crd/src/security.rs [hunk 1]
```diff
@@ -285,49 +249,40 @@ impl KafkaTlsSecurity {
...
-            pod_builder.add_volume(
-                VolumeBuilder::new("server-tls")
-                    .with_empty_dir(Some(""), None)
-                    .build(),
+            // Mounted keystores fore the kafka container
+            cb_kafka.add_volume_mount(
+                Self::STACKABLE_TLS_KEYSTORE_SERVER_DIR_NAME,
+                Self::STACKABLE_TLS_KEYSTORE_SERVER_DIR,
             );
+            pod_builder.add_volume(Self::create_tls_keystore_volume(
```
Comment (sbernauer) [inline rust/crd/src/security.rs:271]: Nit: I would prefer to code to first add the volume and the mount afterwards "to tell a story", but that's just an idea

File: rust/crd/src/security.rs [hunk 2]
```diff
@@ -285,49 +249,40 @@ impl KafkaTlsSecurity {
...
-            cb_prepare
-                .add_volume_mount("internal-tls-mount", Self::STACKABLE_TLS_INTERNAL_MOUNT_DIR);
-            cb_kafka.add_volume_mount("internal-tls-mount", Self::STACKABLE_TLS_INTERNAL_MOUNT_DIR);
-            pod_builder.add_volume(Self::create_tls_volume(
-                "internal-tls-mount",
+            cb_kafka.add_volume_mount(
+                Self::STACKABLE_TLS_KEYSTORE_INTERNAL_DIR_NAME,
+                Self::STACKABLE_TLS_KEYSTORE_INTERNAL_DIR,
+            );
+            pod_builder.add_volume(Self::create_tls_keystore_volume(
```
Comment (sbernauer) [inline rust/crd/src/security.rs:282]: Same here

File: rust/crd/src/lib.rs [hunk 3]
```diff
@@ -353,7 +353,6 @@ impl Storage {
 #[serde(rename_all = "kebab-case")]
 #[strum(serialize_all = "kebab-case")]
 pub enum Container {
-    Prepare,
```
Comment (sbernauer) [inline rust/crd/src/lib.rs:356]: This is actually an breaking CRD change I guess, as you can not configure logging on the prepare container any more.
Is that correct? ;) In that case we should mark it as BREAKING and provide a migration path (delete .spec.foo.logging.prepare)

**Review by sbernauer — COMMENTED** [review]
Awesome, really like this!

File: rust/crd/src/security.rs (hunk 1 above)
Comment (maltesander) [reply rust/crd/src/security.rs]: https://github.com/stackabletech/kafka-operator/pull/621/commits/8a1f1acbad000145fc65ca6b43ee49a6b58a9582

File: rust/crd/src/lib.rs (hunk 3 above)
Comment (maltesander) [reply rust/crd/src/lib.rs:356]: Good point but its not actually breaking (Hashmap -> no CRD change). I grepped through the code to check for any left over "prepare" elements. If you set values for the prepare container they will just be ignored.
Should i explicitly mention that the prepare container was removed in the changelog?

Comment (sbernauer) [reply rust/crd/src/lib.rs:356]: I would have assumed that the valid keys are only the known container as of, rendering currently valid CRs invalid.
https://github.com/stackabletech/operator-rs/blob/b3531bc2922f71f691807053e174404db95f1fb7/src/product_logging/spec.rs#L79

Actually tested it:
```
        apiVersion: kafka.stackable.tech/v1alpha1
        kind: KafkaCluster
        metadata:
          name: simple-kafka
        spec:
          image:
            productVersion: 3.4.0
          clusterConfig:
            zookeeperConfigMapName: xyz
          brokers:
            config:
              logging:
                enableVectorAgent: true
                containers:
                  foo:
                    level: DEBUG
            roleGroups:
              default: {}
```
thread 'tests::test_client_tls' panicked at 'illegal test input: Error("spec.brokers: unknown variant `foo`, expected one of `prepare`, `vector`, `kcat-prober`, `get-service`, `kafka`", line: 12, column: 13)', rust/crd/src/lib.rs:519:63

Comment (sbernauer) [reply rust/crd/src/lib.rs:356]: We *could* keep the enum variant and ignore it for now, but honestly I would just remove it, as probably no one is using it right now and automatic CRD conversion is pretty far away.

Comment (maltesander) [reply rust/crd/src/lib.rs:356]: Not a fan of keeping stuff that does not do anything :)

**Review by sbernauer — CHANGES_REQUESTED** [review]

**Review by sbernauer — DISMISSED** [review]
Thanks!

**Review by sbernauer — APPROVED** [review]

**Pull request merged**